recipes/ffmpeg/all/conanfile.py

```diff
@@ -0,0 +1,352 @@
+            self.requires("openssl/1.1.1k")
+        if self.settings.os == "Linux":
+            if self.options.alsa:
+                self.requires("libalsa/1.1.9")
+            if self.options.xcb:
+                self.requires("xorg/system")
+            if self.options.pulse:
+                self.requires("pulseaudio/13.0")
+
+    def _patch_sources(self):
+        if self._is_msvc and self.options.x264 and not self.options["libx264"].shared:
+            # suppress MSVC linker warnings: https://trac.ffmpeg.org/ticket/7396
+            # warning LNK4049: locally defined symbol x264_levels imported
+            # warning LNK4049: locally defined symbol x264_bit_depth imported
+            tools.replace_in_file(os.path.join(self._source_subfolder, "libavcodec", "libx264.c"),
+                                  "#define X264_API_IMPORTS 1", "")
+        if self.options.ssl == "openssl":
+            # https://trac.ffmpeg.org/ticket/5675
+            openssl_libraries = " ".join(["-l%s" % lib for lib in self.deps_cpp_info["openssl"].libs])
+            tools.replace_in_file(os.path.join(self._source_subfolder, "configure"),
+                                  "check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto -lws2_32 -lgdi32 ||",
+                                  "check_lib openssl openssl/ssl.h OPENSSL_init_ssl %s || " % openssl_libraries)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        args = ["--disable-doc",
+                "--disable-programs"]
+        if self.options.shared:
+            args.extend(["--disable-static", "--enable-shared"])
+        else:
+            args.extend(["--disable-shared", "--enable-static"])
+        args.append("--pkg-config-flags=--static")
+        if self.settings.build_type == "Debug":
+            args.extend(["--disable-optimizations", "--disable-mmx", "--disable-stripping", "--enable-debug"])
+        # since ffmpeg"s build system ignores CC and CXX
+        if "CC" in os.environ:
+            args.append("--cc=%s" % os.environ["CC"])
+        if "CXX" in os.environ:
+            args.append("--cxx=%s" % os.environ["CXX"])
+        if self._is_msvc:
+            args.append("--toolchain=msvc")
+            args.append("--extra-cflags=-%s" % self.settings.compiler.runtime)
+            if int(str(self.settings.compiler.version)) <= 12:
+                # Visual Studio 2013 (and earlier) doesn"t support "inline" keyword for C (only for C++)
+                args.append("--extra-cflags=-Dinline=__inline" % self.settings.compiler.runtime)
+
+        if self.settings.arch == "x86":
+            args.append("--arch=x86")
+
+        if self.options.get_safe("fPIC"):
+            args.append("--enable-pic")
+
+        args.append("--enable-postproc" if self.options.postproc else "--disable-postproc")
```

Comment:
You can rewrite a lot of the options like this (include shared/static):
```python
yes_no = lamda v: "yes" if v else "no"
args = [
    "--enable-zlib={}".format(yes_no(self.options.zlib)),
    "--enable-bzlib={}".format(yes_no(self.options.bzlib)),
    "--enable-lzma={}".format(yes_no(self.options.lzma)),
]
```

This makes much cleaner code.

Reply:
I don't think the `configure` script would accept that:
- some option can be disabled (by appending `--disable-zlib` or `--disable-lzma` for example)
- some option can be enabled (by appending `--enable-libzmq` or `--enable-libx264` for example)

I might be wrong because I'm not familiar with reading big `configure` scripts.

Reply:
I will push and updated version of the recipe managing this way:

```
        if not self.options.postproc:
            args.append("--disable-postproc")
        if not self.options.bzlib:
            args.append("--disable-zlib")
        if not self.options.zlib:
            args.append("--disable-zlib")
        if not self.options.lzma:
            args.append("--disable-lzma")
        if not self.options.iconv:
            args.append("--disable-iconv")
        if self.options.freetype:
            args.append("--enable-libfreetype")
        if self.options.openjpeg:
            args.append("--enable-libopenjpeg")
        if self.options.openh264:
            args.append("--enable-libopenh264")
        if self.options.vorbis:
            args.append("--enable-libvorbis")
        if self.options.opus:
            args.append("--enable-libopus")
        if self.options.zmq:
            args.append("--enable-libzmq")
        if not self.options.sdl2:
            args.append("--disable-sdl2")
        if self.options.x264:
            args.append("--enable-libx264")
        if self.options.x265:
            args.append("--enable-libx265")
        if self.options.vpx:
            args.append("--enable-libvpx")
        if self.options.freetype:
            args.append("--enable-libfreetype")
        if self.options.freetype:
            args.append("--enable-libfreetype")
        if self.options.freetype:
            args.append("--enable-libfreetype")
        if self.options.freetype:
            args.append("--enable-libfreetype")
        if self.options.freetype:
            args.append("--enable-libfreetype")
```

Let me know if you have a cleaner code scheme!